wimm.Guards.Types.Tests/ExtensionTests.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace wimm.Guards.Types.Tests
+{
+    public class ExtensionTests
```

Comment:
It's not clean to me from the test why you use collections and why the interface test is different from the class test.

Reply:
There's no real reason I use collections IEnumerable<T> is just the first interface that comes to mind so I tossed in some lists. Completely arbitrary. 

I did the class test as sort of documentation since all the remarks around the method talk about interfaces and to show that it should still work and I'm not doing something internally that says. 
if Typeof(T).IsAbstract() -> Explode. And to make sure that if I do make that change then I have to change the test + know that it's an  API breaking change.

I guess in short. It doesn't do anything different. Which is exactly what I want to persist (at least for now) and it describes the behaviour of the method.  I'm not hooked on it but I think there's some utility there.